src/jit/codegenarm.cpp

```diff
@@ -1351,12 +1351,41 @@ void CodeGen::genLeaInstruction(GenTreeAddrMode* lea)
 //------------------------------------------------------------------------
 // genSetRegToCond: Generate code to materialize a condition into a register.
 //
+// Arguments:
+//   dstReg - The target register to set to 1 or 0
+//   tree - The GenTree Relop node that was used to set the Condition codes
+//
+// Return Value: none
+//
 // Preconditions:
 //    The condition codes must already have been appropriately set.
 //
 void CodeGen::genSetRegToCond(regNumber dstReg, GenTreePtr tree)
 {
-    NYI("genSetRegToCond");
+    // Emit code like that:
+    //   ...
+    //   bgt True
+    //   movs rD, #0
+    //   b Next
+    // True:
+    //   movs rD, #1
+    // Next:
+    //   ...
```

Comment:
These conditional moves are required to be governed by additional `If-Then` instruction on Thumb. For example LLVM emits this:
```
   0:	280d      	cmp	r0, #13
   2:	bfd4      	ite	le
   4:	2200      	movle	r2, #0
   6:	2201      	movgt	r2, #1
```
However there are some comments regarding these `IT` instructions at the `emitarm.cpp`:
```c++
#ifdef FEATURE_ITINSTRUCTION

/*****************************************************************************
 * The "IT" instruction is deprecated (with a very few exceptions). Don't generate it!
 * Don't delete this code, though, in case we ever want to bring it back.
 *****************************************************************************/
```
What is the reason `IT` instructions were deprecated?